lib/commands/project/project.js

```diff
@@ -363,3 +373,57 @@ function spawn(command, options, cwd, cb) {
     cb(err);
   });
 }
+
+
+//.option('-p, --path-name [path]', 'a sepecific path of the api')
+//.option('-f, --test-module <module>', 'one of: ' + testmodules)
+//.option('-t, --assertion-format <type>', 'one of: ' + assertiontypes)
+function testGenerate(directory, options, cb) {
+  var pathList = [];
+  var desiredPaths = [];
+  var swaggerSpec = require('../../util/spec');
+
+  //read the yaml file and validate it
+  readProject(directory, options, function(err, project) {
+    if (err) { return cb(err); }
+    swaggerSpec.validateSwagger(project.api.swagger, options, cb);
```

Comment:
put all your code from here in the callback function and call `cb` when everything is done.